copier/config/user_data.py

```diff
@@ -120,12 +154,21 @@ def query_user_data(
         "str": str,
         "yaml": parse_yaml_string,
     }
-    result: AnyByStrDict = {}
+    env = get_env(
+        envops=envops, filters={"to_nice_yaml": to_nice_yaml}, undefined=StrictUndefined
```

Comment:
Still passing `undefined=StrictUndefined` in kwargs I didn't want to make this behavior the default. But maybe we should? Not sure.

Reply:
It would be nice to be able to configure undefined behavior in `_envops`, though. But that can be a separate improvement.

I'd like copier to be widely used outside of python projects.

I guess using more relaxed checks makes it easier for templaters. So, unless there's a clear advantage, I'd vote against being strict. Also, having different behaviors depending on the context might be more confusing to the templater.

So I'd just leave the same undefined behavior as we had until now.

About deduplicating the addition of `to_nice_yaml`, go ahead. :ok_hand:

Reply:
OK, let's leave it non-strict, and add a strict option to EnvOps in another PR :slightly_smiling_face: I'll have to comment out the test that was checking for undefined variables.